sunkit_image/coalignment_module/match_template.py

```diff
@@ -0,0 +1,210 @@
+    return numerator / denominator
+
+
+def _get_correlation_shifts(array):
+    """
+    Calculate the shifts in x and y directions based on the correlation array.
+
+    Parameters
+    ----------
+    array : `numpy.ndarray`
+        A 2D array representing the correlation values.
+
+    Returns
+    -------
+    tuple
+        The shifts in y and x directions.
+
+    Raises
+    ------
+    ValueError
+        If the input array dimensions are greater than 3 in any direction.
+    """
+    ny, nx = array.shape
+    if nx > 3 or ny > 3:
+        msg = "Input array dimension should not be greater than 3 in any dimension."
+        raise ValueError(msg)
+
+    ij = np.unravel_index(np.argmax(array), array.shape)
+    x_max_location, y_max_location = ij[::-1]
+
+    y_location = _parabolic_turning_point(array[:, x_max_location]) if ny == 3 else 1.0 * y_max_location
+    x_location = _parabolic_turning_point(array[y_max_location, :]) if nx == 3 else 1.0 * x_max_location
+
+    return y_location, x_location
+
+
+def _find_best_match_location(corr):
+    """
+    Find the best match location in the correlation array.
+
+    Parameters
+    ----------
+    corr : `numpy.ndarray`
+        The correlation array.
+
+    Returns
+    -------
+    tuple
+        The best match location in the y and x directions.
+    """
+    ij = np.unravel_index(np.argmax(corr), corr.shape)
+    cor_max_x, cor_max_y = ij[::-1]
+
+    array_maximum = corr[
+        max(0, cor_max_y - 1) : min(cor_max_y + 2, corr.shape[0] - 1),
+        max(0, cor_max_x - 1) : min(cor_max_x + 2, corr.shape[1] - 1),
+    ]
+
+    y_shift_maximum, x_shift_maximum = _get_correlation_shifts(array_maximum)
+
+    y_shift_correlation_array = y_shift_maximum + cor_max_y
+    x_shift_correlation_array = x_shift_maximum + cor_max_x
+
+    return y_shift_correlation_array, x_shift_correlation_array
+
+
+@register_coalignment_method("match_template")
+def match_template_coalign(input_array, template_array):
+    """
+    Perform coalignment by matching the template array to the input array.
+
+    Parameters
+    ----------
+    input_array : `numpy.ndarray`
+        The input 2D array to be coaligned.
+    template_array : `numpy.ndarray`
+        The template 2D array to align to.
+
+    Returns
+    -------
+    tuple
+        A tuple where the first element is the coaligned array and the second element is another tuple containing the
+        shifts in x and y directions (x_shift, y_shift),
+    """
+    corr = match_template(np.float64(input_array), np.float64(template_array))
+
+    # Find the best match location
+    y_shift, x_shift = _find_best_match_location(corr)
+    # Calculate the clipping required
+    yclips, xclips = _calculate_clipping(x_shift * u.pix, y_shift * u.pix)
+    # Clip 'em
+    coaligned_target_array = _clip_edges(input_array, yclips, xclips)
+    # Apply the shift to get the coaligned input array
+    return (coaligned_target_array, (x_shift, y_shift))
```

Comment:
Now, we'll want to just return the `x_shift` and `y_shift` but we should do this in the format of the affine transform components, i.e. scale, rotation, shift. So in this case, scale would be [1,1], rotation would be the identity, and the shift would be [x_shift, y_shift]